samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -265,6 +267,11 @@ public final void aggregateEndOfStream(EndOfStreamMessage eos, SystemStreamParti
     SystemStream stream = ssp.getSystemStream();
     if (eosStates.isEndOfStream(stream)) {
       LOG.info("Input {} reaches the end for task {}", stream.toString(), taskName.getTaskName());
+      if (eos.getTaskName() != null) {
```

Comment:
I don't understand this condition? In what case task name is null? And why we send broadcast in case it is not NULL?

Reply:
Chatted offline. This check verifies this is the aggregation task and needs to broadcast the result to other partitions. Added the comment.